compliance/collection/metrics/metrics.go

```diff
@@ -125,17 +164,48 @@ func ObserveNodeInventoryScan(inventory *storage.NodeInventory) {
 		rhelPackageCount = len(components.GetRhelComponents())
 	}
 	numberOfRHELPackages.With(prometheus.Labels{
-		"node_name":    inventory.GetNodeName(),
-		"os_namespace": components.GetNamespace(),
+		"node_name":       inventory.GetNodeName(),
+		"os_namespace":    components.GetNamespace(),
+		"scanner_version": ScannerVersionV2,
 	}).Set(float64(rhelPackageCount))
 
 	rhelContentSets := 0
 	if components.GetRhelContentSets() != nil {
 		rhelContentSets = len(components.GetRhelContentSets())
 	}
 	numberOfContentSets.With(prometheus.Labels{
-		"node_name":    inventory.GetNodeName(),
-		"os_namespace": components.GetNamespace(),
+		"node_name":       inventory.GetNodeName(),
+		"os_namespace":    components.GetNamespace(),
+		"scanner_version": ScannerVersionV2,
+	}).Set(float64(rhelContentSets))
+}
+
+// ObserveNodeIndexReport observes the metric for Scanner v4.
+func ObserveNodeIndexReport(report *v4.IndexReport, nodeName string) {
+	rhelPackageCount := 0
+	contents := report.GetContents()
+
+	if contents == nil {
+		return
+	}
+
+	if contents.GetPackages() != nil {
+		rhelPackageCount = len(contents.GetPackages())
+	}
```

Comment:
`GetPackages` returns nil only when contents is nil. Not sure whether we need that check, but it can stay.